homeassistant/components/dyson/sensor.py

```diff
@@ -210,11 +202,9 @@ def __init__(self, device):
     @property
     def state(self):
         """Return Humidity value."""
-        if self._device.environmental_state:
-            if self._device.environmental_state.humidity == 0:
-                return STATE_OFF
-            return self._device.environmental_state.humidity
-        return None
+        if self._device.environmental_state.humidity == 0:
+            return STATE_OFF
```

Comment:
What does off mean here? Does it mean that the sensor is unavailable? May that change during entity lifetime? We should probably use `available` property instead.

Reply:
According to the library code, `environmental_state.humidity` is set to 0 when the device reports a humidity "OFF". I believe it means that the fan is currently turned of without the continuous monitoring activated. Therefore, it just stop reporting humidity and a bunch of other environmental states. So in this case, should we report the sensor as unavailable?

Code here: https://github.com/etheralm/libpurecool/blob/e797babbcb03dd6b0a9fb9b471087ff1ad1885eb/libpurecool/dyson_pure_state.py#L114-L124

Reply:
If the device is actually reporting off, it may be best to keep the state as off, as we have it now. Unavailable should mean that we know that the device failed to report or can't report its state.